### DISCUSSION TITLE
Saving definitions to workflow definition store

### DISCUSSION THREAD

**Discussion by rmanjusha** [elsa-workflows/elsa-core#5434] [Q&A]
How to save JSON  workflow definitions files from physical folder into Entity framework workflow definition store?

**Comment by glime-ai** [comment]
To save JSON workflow definitions from a physical folder into the Entity Framework workflow definition store in Elsa, you would typically follow these steps:

1. Load the JSON files from the physical folder.
2. Deserialize the JSON content into `WorkflowDefinition` objects.
3. Use the `IWorkflowDefinitionStore` service to save the `WorkflowDefinition` objects into the database.

Here's a high-level example of how you might write the code to accomplish this:

```csharp
// Assuming you have a service that can read and deserialize JSON files into WorkflowDefinition objects
var workflowDefinitions = myWorkflowDefinitionService.LoadWorkflowDefinitionsFromFolder("path/to/your/json/folder");

// Get an instance of IWorkflowDefinitionStore from the DI container
var workflowDefinitionStore = serviceProvider.GetRequiredService<IWorkflowDefinitionStore>();

// Save each workflow definition into the database
foreach (var workflowDefinition in workflowDefinitions)
{
    await workflowDefinitionStore.SaveAsync(workflowDefinition, cancellationToken);
}
```

In the context provided, you can use the `JsonWorkflowProvider` class example from the GitHub discussion as a starting point to load the JSON files and deserialize them into `WorkflowDefinitionVersion` objects. You would then map these to `WorkflowDefinition` objects as required by Elsa 3.

Please note that the exact implementation details, such as the method names and parameters, may vary depending on the version of Elsa you are using and the specific setup of your application. You may need to adjust the code to fit your context.

If you need further assistance with the code or have specific questions about the implementation, please let me know, and I can provide more detailed guidance.

This comment was generated by [Glime](https://glimelab.ai/?utm_source=dRbvTb7IhCSYiI&utm_medium=github).
